File: typescript/tsfind/tests/fileresult.test.ts
/*
 * fileresult.test.js
 *
 * Some tests of fileresult.js
 */

import {FileType} from '../src/filetype';
import {FileResult} from '../src/fileresult';

describe('testing fileresult', () => {
    it('testFileResultAbsPath', () => {
        const filePath = `${process.env.HOME}/src/xfind/javascript/jsfind/src/fileresult.js`;
        const fileResult = new FileResult(filePath, FileType.Code, 0, 0);
        const expected = `${process.env.HOME}/src/xfind/javascript/jsfind/src/fileresult.js`;
        expect(fileResult.filePath).toEqual(expected);
    });

    it('testFileResultTildePath', () => {
        const filePath = '~/src/xfind/javascript/jsfind/src/fileresult.js';
        const fileResult = new FileResult(filePath, FileType.Code, 0, 0);
        const expected = '~/src/xfind/javascript/jsfind/src/fileresult.js';
        expect(fileResult.filePath).toEqual(expected);
    });

    it('testFileResultRelPath1', () => {
        const filePath = './fileresult.js';
        const fileResult = new FileResult(filePath, FileType.Code, 0, 0);
        const expected = './fileresult.js';
        expect(fileResult.filePath).toEqual(expected);
    });

    it('testFileResultRelPath2', () => {
        const filePath = './fileresult.js';
        const fileResult = new FileResult(filePath, FileType.Code, 0, 0);
        const expected = './fileresult.js';
        expect(fileResult.filePath).toEqual(expected);
    });

    it('testFileResultRelPath3', () => {
        const filePath = '../fileresult.js';
        const fileResult = new FileResult(filePath, FileType.Code, 0, 0);
        const expected = '../fileresult.js';
        expect(fileResult.filePath).toEqual(expected);
    });
});

File: dart/dartfind/test/file_result_test.dart
import 'dart:io' show File, Platform;

import 'package:dartfind/dartfind.dart' show FileType, FileResult;
import 'package:test/test.dart';

void main() {
  test('test FileResult with absolute path', () {
    var path =
        '${Platform.environment['HOME']}/src/xfind/dart/dartfind/lib/src/file_result.dart';
    var fr = FileResult(File(path), FileType.code, null);
    expect(fr.toString(), path);
  });

  test('test FileResult with tilde path', () {
    var path = '~/src/xfind/dart/dartfind/lib/src/file_result.dart';
    var fr = FileResult(File(path), FileType.code, null);
    expect(fr.toString(), path);
  });

  test('test FileResult with rel path #1', () {
    var path = './file_result.dart';
    var fr = FileResult(File(path), FileType.code, null);
    expect(fr.toString(), path);
  });

  test('test FileResult with rel path #2', () {
    var path = './file_result.dart';
    var fr = FileResult(File(path), FileType.code, null);
    expect(fr.toString(), path);
  });
}
